File: solutions/csharp/rotational-cipher/1/RotationalCipher.cs
using System.Text;

public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
    {
        StringBuilder result = new StringBuilder();
        foreach (char ch in text)
        {
            if (Char.IsLetter(ch))
            {
                char bas = Char.IsUpper(ch) ? 'A' : 'a';
                result.Append((char)((ch - bas + shiftKey) % 26 + bas));
            }
            else
                result.Append(ch);
        }
        return result.ToString();
    }
}

File: solutions/cpp/rotational-cipher/1/rotational_cipher.cpp
#include "rotational_cipher.h"
#include <string>

namespace rotational_cipher {

std::string rotate(std::string text, int key) {
    std::string result;
    for (char c : text) {
        if (c >= 'a' && c <= 'z') result += 'a' + ((c + key) - 'a') % 26;
        else if (c >= 'A' && c <= 'Z') result += 'A' + ((c + key) - 'A') % 26;
        else result += c;
    }
    return result;
}

}  // namespace rotational_cipher
